Block/Service/AbstractCollectionsBlockService.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+namespace Sonata\ClassificationBundle\Block\Service;
+
+use Sonata\AdminBundle\Form\FormMapper;
+use Sonata\BlockBundle\Block\BlockContextInterface;
+use Sonata\BlockBundle\Model\BlockInterface;
+use Sonata\ClassificationBundle\Admin\CollectionAdmin;
+use Sonata\ClassificationBundle\Entity\CollectionManager;
+use Sonata\ClassificationBundle\Model\CollectionInterface;
+use Sonata\CoreBundle\Model\Metadata;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+abstract class AbstractCollectionsBlockService extends BaseClassificationBlockService
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function execute(BlockContextInterface $blockContext, Response $response = null)
+    {
+        $collection  = $this->getCollection($blockContext->getSetting('collectionId'), $blockContext->getSetting('collection'));
+        $context     = $blockContext->getSetting('context');
+
+        $criteria = array(
+            'enabled' => true,
+            'context' => $context,
+        );
```

Comment:
Several of them. If you need variables for expressing something more, consider comments instead